From d3d30c4286d646946cecffb99286c3ce1503015c Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Fri, 14 Feb 2014 15:35:11 -0500
Subject: [PATCH] gsocket: fix g_socket_condition_timed_wait() recovery after
 EINTR

After getting an EINTR, g_socket_condition_timed_wait() has to adjust
its timeout, but it was trying to convert from nanoseconds to
microseconds by multiplying by 1000 rather than dividing... Oops.

https://bugzilla.gnome.org/show_bug.cgi?id=724239
---
 gio/gsocket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gsocket.c b/gio/gsocket.c
index fb3bf96..72f2470 100644
--- a/gio/gsocket.c
+++ b/gio/gsocket.c
@@ -3662,7 +3662,7 @@ g_socket_condition_timed_wait (GSocket       *socket,
 
 	if (timeout != -1)
 	  {
-	    timeout -= (g_get_monotonic_time () - start_time) * 1000;
+	    timeout -= (g_get_monotonic_time () - start_time) / 1000;
 	    if (timeout < 0)
 	      timeout = 0;
 	  }
-- 
1.8.5.3